include/lsst/afw/geom/SphPoint.h

```diff
@@ -0,0 +1,319 @@
+         * floating-point fields are exactly equal.
+         *
+         * @param other the point to test for equality
+         * @return true if this point matches @c other exactly, false otherwise
+         *
+         * @par Exception Safety:
+         * This operator shall not throw exceptions.
+         */
+        bool operator==(SphPoint const & other) const noexcept;
+
+        /**
+         * @brief Tests whether two points represent the same position.
+         *
+         * To make equality transitive (as well as consistent with built-in
+         * types), two objects are considered equal if and only if their
+         * floating-point fields are exactly equal.
+         *
+         * @param other the point to test for equality
+         * @return true if this point does not match @c other exactly, false otherwise
+         *
+         * @par Exception Safety:
+         * This operator shall not throw exceptions.
+         */
+        bool operator!=(SphPoint const & other) const noexcept;
+
+        /**
+         * @brief Direction from one point to another.
+         *
+         * This method finds the shortest (great-circle) arc between two
+         * points, and characterizes its direction by the angle between
+         * it and a line of latitude passing through this point. 0 represents
+         * due east, &pi;/2 represents due north. If the points are on
+         * opposite sides of the sphere, the bearing may be any value.
+         *
+         * @param other the point to which to measure the bearing
+         * @return the direction, as defined above, in the interval [0, 2&pi;).
+         *
+         * @throws DomainError Thrown if this point is either coordinate pole.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         *
+         * @note For two points @c A and @c B, <tt>A.bearingTo(B)</tt> will in
+         *       general not be 180 degrees away from <tt>B.bearingTo(A)</tt>
+         */
+        Angle bearingTo(SphPoint const & other) const;
+
+        /**
+         * @brief Angular distance between two points.
+         *
+         * @param other the point to which to measure the separation
+         * @return the length of the shortest arc between the two points. Shall
+         *         not be negative.
+         *
+         * @par Exception Safety:
+         * This method shall not throw exceptions.
+         */
+        Angle separationTo(SphPoint const & other) const noexcept;
+
+        /*
+         * Transformations of points
+         */
+
+        /**
+         * @brief A point defined by rotating this point around an arbitrary axis.
+         *
+         * @param axis the point around which to perform the rotation.
+         * @param amount the angle by which to rotate around @c axis.
+         *               Positive angles represent counterclockwise
+         *               rotations.
```

Comment:
I find the two parameter descriptions unclear. What does it mean to rotate around a point? Perhaps something like this might be clearer, though I hope it may give you an idea on how to do even better:
- @brief axis  A point defining the Z axis of rotation frame
- @brief amount  The amount of rotation, where positive values rotate from the X axis to Y axis of the rotation frame